### PR TITLE
Story 3 01 & Refactor & checkStyle satisfaction

### PR DESCRIPTION
**Pull request by Crutovel**
Student => scripts schema.sql and data.sql, Entity, dto Resource, HATEOAS Assembler, Repository, Service, Controller.
Entities => Added or modified equals(), hashcode(), toString().
A lot of refactored stuff (useless imports, constructors, changed @RequestMapping => @{Method}Mapping etc.)

### CODE REVIEW

File: src/main/java/com/softserve/teamproject/repository/StudentRepository.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.softserve.teamproject.repository;
+
+import com.softserve.teamproject.entity.Student;
+import com.softserve.teamproject.repository.custom.StudentRepositoryCustom;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.querydsl.QueryDslPredicateExecutor;
+
+public interface StudentRepository extends JpaRepository<Student, Integer>,
+    QueryDslPredicateExecutor, StudentRepositoryCustom {
```
Comment (thedikiy) [inline src/main/java/com/softserve/teamproject/repository/StudentRepository.java]: QueryDsl `<Student>`

File: src/main/java/com/softserve/teamproject/service/impl/StudentServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+   * Gets all students.
+   *
+   * @return all students
+   */
+  public Iterable<StudentResource> getAllStudents() {
+    return convertToResource(studentRepository.findAll());
+  }
+
+  private Iterable<StudentResource> convertToResource(Iterable<Student> events) {
+    List<StudentResource> studentResources = new ArrayList<>();
```
Comment (thedikiy) [inline src/main/java/com/softserve/teamproject/service/impl/StudentServiceImpl.java]: events?

File: src/main/java/com/softserve/teamproject/entity/Group.java [hunk 3]
```diff
@@ -182,25 +187,18 @@ public int hashCode() {
...
+    return "Group{"
+        + "id=" + id
+        + ", name='" + name + '\''
+        + ", teachers=" + teachers
+        + ", location=" + location.getName()
+        + ", startDate=" + startDate
+        + ", finishDate=" + finishDate
+        + ", status=" + status
+        + ", specialization=" + specialization
+        + ", budgetOwner=" + budgetOwner
```
Comment (thedikiy) [inline src/main/java/com/softserve/teamproject/entity/Group.java]: I feel like budgetOwner.getName() would be better, same for status and specialization

Comment (Crutovel) [inline src/main/java/com/softserve/teamproject/entity/Group.java]: i agree

File: src/main/java/com/softserve/teamproject/repository/StudentRepository.java (hunk 1 above)
Comment (Crutovel) [inline src/main/java/com/softserve/teamproject/repository/StudentRepository.java]: thx

File: src/main/java/com/softserve/teamproject/service/impl/StudentServiceImpl.java (hunk 2 above)
Comment (Crutovel) [inline src/main/java/com/softserve/teamproject/service/impl/StudentServiceImpl.java]: yes sure

**Pull request merged**